src/PhpCsFixer/PhpCsFixerRuleSet.php

```diff
@@ -193,5 +209,122 @@ public static function getRules(): array
             'whitespace_after_comma_in_array' => true,
             'declare_strict_types' => true,
         ];
+
+        // Set the header dynamically based on the current detected project name.
+        $projectLicense = self::detectCentreonProjectLicense(__DIR__);
+        $phpLicenseHeader = match ($projectLicense) {
+            self::LICENSE_APACHE => self::getHeaderLicenseApache(self::YEAR),
+            self::LICENSE_PRIVATE => self::getHeaderLicensePrivate(self::YEAR),
+            default => '',
+        };
+        if ($phpLicenseHeader) {
+            $rules += [
+                'header_comment' => [
+                    'location' => 'after_open',
+                    'header' => $phpLicenseHeader,
+                ],
+            ];
+        }
+
+        return $rules;
+    }
+
+    /**
+     * Return the header for an Apache License.
+     *
+     * @param int $year
+     */
+    private static function getHeaderLicenseApache(int $year): string
+    {
+        return <<<HEADER
```

Comment:
Maybe the header content should be in a separate file?

Reply:
We can do that.
At this point, for the year, either I remove the const and hard code it in the file, or I use a pattern replacement.
I would prefer the hard date into a such file. WDYT ?

Reply:
Your proposal to use a replacement pattern in the file is a good idea